// @uploadcare/blocks/index.ssr.js
export const ActivityBlock = class {
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const ActivityHeader = class {
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const BaseComponent = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Block = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const CameraSource = class {
  static template = `
  <lr-activity-header>
    <button type="button" class="mini-btn" set="onclick: *historyBack">
      <lr-icon name="back"></lr-icon>
    </button>
    <div set="@hidden: !cameraSelectHidden">
      <lr-icon name="camera"></lr-icon>
      <span l10n="caption-camera"></span>
    </div>
    <lr-select
      class="camera-select"
      set="$.options: cameraSelectOptions; @hidden: cameraSelectHidden; onchange: onCameraSelectChange"
    >
    </lr-select>
    <button type="button" class="mini-btn close-btn" set="onclick: *closeModal">
      <lr-icon name="close"></lr-icon>
    </button>
  </lr-activity-header>
  <div class="content">
    <video
      autoplay
      playsinline
      set="srcObject: video; style.transform: videoTransformCss; @hidden: videoHidden"
      ref="video"
    ></video>
    <div class="message-box" set="@hidden: messageHidden">
      <span l10n="l10nMessage"></span>
      <button
        type="button"
        set="onclick: onRequestPermissions; @hidden: requestBtnHidden"
        l10n="camera-permissions-request"
      ></button>
    </div>
    <button type="button" class="shot-btn" set="onclick: onShot; @disabled: shotBtnDisabled">
      <lr-icon name="camera"></lr-icon>
    </button>
  </div>
`;
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const CloudImageEditor = class {
  static className = `cloud-image-editor`;
  static template = `
  <svg width='0' height='0' style='position:absolute'><symbol viewBox='0 0 20 20' id='uc-icon-brightness' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M15 10a5 5 0 0 1-5 5m5-5a5 5 0 0 0-5-5m5 5h-5m0 5a5 5 0 0 1 0-10m0 10V5m0 15v-3M2.93 2.929 5.05 5.05M0 10h3m-.07 7.071 2.12-2.121M10 0v3m7.07 14.071-2.12-2.121M20 10h-3m.07-7.071L14.95 5.05m-.626 2.45H10m4.324 5H10'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-closeMax' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='m3 3 14 14m0-14L3 17'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-contrast' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M2 10a8 8 0 1 0 16 0 8 8 0 1 0-16 0m8-8v16m8-8h-8m7.598 2.5H10m6.24 2.5H10m7.6-7.5H10M16.242 5H10'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-crop' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M20 14H7.005C6.45 14 6 13.55 6 12.995V0M0 6h13.067c.515 0 .933.418.933.933V20M14.5.4 13 2l1.5 1.6M13 2h2a3 3 0 0 1 3 3v2M5.5 19.6 7 18l-1.5-1.6M7 18H5a3 3 0 0 1-3-3v-2'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-done' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='m1 10.632 5.684 5.684L19 4'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-enhance' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M19 13h-2m0 0a4 4 0 0 1-4-4m4 4a4 4 0 0 0-4 4m0-8V7m0 2a4 4 0 0 1-4 4m-2 0h2m0 0a4 4 0 0 1 4 4m0 0v2M8 8.5H6.5m0 0a2 2 0 0 1-2-2m2 2a2 2 0 0 0-2 2m0-4V5m0 1.5a2 2 0 0 1-2 2M1 8.5h1.5m0 0a2 2 0 0 1 2 2m0 0V12M12 3h-1m0 0a1 1 0 0 1-1-1m1 1a1 1 0 0 0-1 1m0-2V1m0 1a1 1 0 0 1-1 1M8 3h1m0 0a1 1 0 0 1 1 1m0 0v1'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-exposure' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M10 20v-3M2.93 2.929 5.05 5.05M0 10h3m-.07 7.071 2.12-2.121M10 0v3m7.07 14.071-2.12-2.121M20 10h-3m.07-7.071L14.95 5.05M5 10a5 5 0 1 0 10 0 5 5 0 1 0-10 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-filters' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M4.5 6.5a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0m-3.5 6a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0m7 0a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-flip' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M19.6 5 18 3.5 16.4 5m3.2 10L18 16.5 16.4 15M18 3.523v12.954M3.3 8.5h10.654c.301 0 .415-.395.159-.554L3.459 1.286A.3.3 0 0 0 3 1.542V8.2a.3.3 0 0 0 .3.3zm0 3h10.654c.301 0 .415.395.159.554l-10.654 6.66A.3.3 0 0 1 3 18.458v-6.66a.3.3 0 0 1 .3-.3z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-gamma' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M17 3C9 6 2.5 11.5 2.5 17.5m0 0h1m-1 0v-1m14 1h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3-14v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-mirror' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M5 .4 3.5 2 5 3.6M15 .4 16.5 2 15 3.6M3.52 2h12.957M8.5 16.7V6.046c0-.301-.394-.415-.554-.159L1.287 16.541a.3.3 0 0 0 .255.459H8.2a.3.3 0 0 0 .3-.3zm3 0V6.046c0-.301.395-.415.555-.159l6.659 10.654a.3.3 0 0 1-.255.459H11.8a.3.3 0 0 1-.3-.3z'/></symbol><symbol viewBox='0 0 40 40' id='uc-icon-original' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M0 40 40 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-rotate' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M13.5.4 12 2l1.5 1.6M12.023 2H14.4A3.6 3.6 0 0 1 18 5.6V8M4 17h9a1 1 0 0 0 1-1V7a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-sad' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M2 17c4.418-4 11.582-4 16 0M16.5 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm-11 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-saturation' xmlns='http://www.w3.org/2000/svg'><defs><linearGradient id='ouc-icon-id__a' x1='10.001' y1='1' x2='10.001' y2='19' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient></defs><circle cx='10.001' cy='10' r='9' transform='rotate(90 10 10)' fill='url(#ouc-icon-id__a)'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-slider' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M0 10h11m0 0a2 2 0 1 0 4 0m-4 0a2 2 0 1 1 4 0m0 0h5'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-tuning' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M8 10h11M1 10h4M1 4.5h11m3 0h4m-18 11h11m3 0h4m-7-11a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0M5 10a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0m7 5.5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-vibrance' xmlns='http://www.w3.org/2000/svg'><path d='M2.125 5.64A8.96 8.96 0 0 0 1.001 10a8.96 8.96 0 0 0 1.124 4.36V5.64z' fill='url(#ruc-icon-id__a)'/><path d='M2.875 15.499V4.502a9.053 9.053 0 0 1 1.75-1.72v14.437a9.05 9.05 0 0 1-1.75-1.72z' fill='url(#ruc-icon-id__b)'/><path d='M5.375 17.722c.548.33 1.134.601 1.75.809V1.469a8.956 8.956 0 0 0-1.75.81v15.443z' fill='url(#ruc-icon-id__c)'/><path d='M7.875 1.253v17.495c.564.136 1.15.22 1.75.244V1.008a9 9 0 0 0-1.75.245z' fill='url(#ruc-icon-id__d)'/><path d='M10.375 1.008v17.984a9 9 0 0 0 1.75-.244V1.252a9 9 0 0 0-1.75-.244z' fill='url(#ruc-icon-id__e)'/><path d='M12.875 1.469V18.53a8.957 8.957 0 0 0 1.75-.808V2.277a8.957 8.957 0 0 0-1.75-.808z' fill='url(#ruc-icon-id__f)'/><path d='M15.375 2.78v14.44a9.053 9.053 0 0 0 1.75-1.72v-11a9.054 9.054 0 0 0-1.75-1.72z' fill='url(#ruc-icon-id__g)'/><path d='M17.875 5.638v8.724A8.959 8.959 0 0 0 19.001 10a8.96 8.96 0 0 0-1.126-4.362z' fill='url(#ruc-icon-id__h)'/><defs><linearGradient id='ruc-icon-id__a' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__b' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__c' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__d' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__e' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__f' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__g' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__h' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient></defs></symbol><symbol viewBox='0 0 20 20' id='uc-icon-warmth' xmlns='http://www.w3.org/2000/svg'><path d='m7.5 13.05.429.42.171-.175v-.244h-.6zm5 0h-.6v.245l.172.175.428-.42zM8.1 3.5c0-1.05.85-1.9 1.9-1.9V.4a3.1 3.1 0 0 0-3.1 3.1h1.2zm0 9.55V3.5H6.9v9.55h1.2zm-1 2.45c0-.79.315-1.506.829-2.03l-.858-.84A4.088 4.088 0 0 0 5.9 15.5h1.2zm2.9 2.9a2.9 2.9 0 0 1-2.9-2.9H5.9a4.1 4.1 0 0 0 4.1 4.1v-1.2zm2.9-2.9a2.9 2.9 0 0 1-2.9 2.9v1.2a4.1 4.1 0 0 0 4.1-4.1h-1.2zm-.829-2.03c.514.524.829 1.24.829 2.03h1.2c0-1.117-.447-2.13-1.171-2.87l-.858.84zM11.9 3.5v9.55h1.2V3.5h-1.2zM10 1.6c1.05 0 1.9.85 1.9 1.9h1.2A3.1 3.1 0 0 0 10 .4v1.2z' fill='currentColor'/><path d='M10 14V8' stroke='currentColor' stroke-width='1.2' stroke-linecap='round'/><path d='M14 3h3m-3 3h3m-3 3h3m-8 6.5a1 1 0 1 0 2 0 1 1 0 1 0-2 0' stroke='currentColor' stroke-width='1.2'/></symbol></svg>
  <div class="wrapper wrapper_desktop">
    <lr-presence-toggle class="network_problems_splash" set="visible: presence.networkProblems;">
      <div class="network_problems_content">
        <div class="network_problems_icon">
          <lr-icon name="sad"></lr-icon>
        </div>
        <div class="network_problems_text">Network error</div>
      </div>
      <div class="network_problems_footer">
        <lr-btn-ui theme="primary" text="Retry" set="onclick: *on.retryNetwork"></lr-btn-ui>
      </div>
    </lr-presence-toggle>
    <div class="viewport">
      <div class="file_type_outer">
        <div class="file_type">{{fileType}}</div>
      </div>
      <div class="image_container" ref="img-container-el">
        <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=" class="image image_visible_from_editor" ref="img-el" />
        <lr-editor-image-cropper ref="cropper-el"></lr-editor-image-cropper>
        <lr-editor-image-fader ref="fader-el"></lr-editor-image-fader>
      </div>
      <div class="info_pan">{{msg}}</div>
    </div>
    <div class="toolbar">
      <lr-line-loader-ui set="active: showLoader"></lr-line-loader-ui>
      <div class="toolbar_content toolbar_content__editor">
        <lr-editor-toolbar></lr-editor-toolbar>
      </div>
    </div>
  </div>
`;
  static observedAttributes = ["uuid", "cdn-url", "crop-preset", "tabs"];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const CloudImageEditorActivity = class {
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const CloudImageEditorBlock = class {
  static className = `cloud-image-editor`;
  static template = `
  <svg width='0' height='0' style='position:absolute'><symbol viewBox='0 0 20 20' id='uc-icon-brightness' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M15 10a5 5 0 0 1-5 5m5-5a5 5 0 0 0-5-5m5 5h-5m0 5a5 5 0 0 1 0-10m0 10V5m0 15v-3M2.93 2.929 5.05 5.05M0 10h3m-.07 7.071 2.12-2.121M10 0v3m7.07 14.071-2.12-2.121M20 10h-3m.07-7.071L14.95 5.05m-.626 2.45H10m4.324 5H10'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-closeMax' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='m3 3 14 14m0-14L3 17'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-contrast' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M2 10a8 8 0 1 0 16 0 8 8 0 1 0-16 0m8-8v16m8-8h-8m7.598 2.5H10m6.24 2.5H10m7.6-7.5H10M16.242 5H10'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-crop' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M20 14H7.005C6.45 14 6 13.55 6 12.995V0M0 6h13.067c.515 0 .933.418.933.933V20M14.5.4 13 2l1.5 1.6M13 2h2a3 3 0 0 1 3 3v2M5.5 19.6 7 18l-1.5-1.6M7 18H5a3 3 0 0 1-3-3v-2'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-done' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='m1 10.632 5.684 5.684L19 4'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-enhance' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M19 13h-2m0 0a4 4 0 0 1-4-4m4 4a4 4 0 0 0-4 4m0-8V7m0 2a4 4 0 0 1-4 4m-2 0h2m0 0a4 4 0 0 1 4 4m0 0v2M8 8.5H6.5m0 0a2 2 0 0 1-2-2m2 2a2 2 0 0 0-2 2m0-4V5m0 1.5a2 2 0 0 1-2 2M1 8.5h1.5m0 0a2 2 0 0 1 2 2m0 0V12M12 3h-1m0 0a1 1 0 0 1-1-1m1 1a1 1 0 0 0-1 1m0-2V1m0 1a1 1 0 0 1-1 1M8 3h1m0 0a1 1 0 0 1 1 1m0 0v1'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-exposure' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M10 20v-3M2.93 2.929 5.05 5.05M0 10h3m-.07 7.071 2.12-2.121M10 0v3m7.07 14.071-2.12-2.121M20 10h-3m.07-7.071L14.95 5.05M5 10a5 5 0 1 0 10 0 5 5 0 1 0-10 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-filters' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M4.5 6.5a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0m-3.5 6a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0m7 0a5.5 5.5 0 1 0 11 0 5.5 5.5 0 1 0-11 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-flip' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M19.6 5 18 3.5 16.4 5m3.2 10L18 16.5 16.4 15M18 3.523v12.954M3.3 8.5h10.654c.301 0 .415-.395.159-.554L3.459 1.286A.3.3 0 0 0 3 1.542V8.2a.3.3 0 0 0 .3.3zm0 3h10.654c.301 0 .415.395.159.554l-10.654 6.66A.3.3 0 0 1 3 18.458v-6.66a.3.3 0 0 1 .3-.3z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-gamma' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M17 3C9 6 2.5 11.5 2.5 17.5m0 0h1m-1 0v-1m14 1h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3 0h1m-3-14v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1m0 3v-1'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-mirror' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M5 .4 3.5 2 5 3.6M15 .4 16.5 2 15 3.6M3.52 2h12.957M8.5 16.7V6.046c0-.301-.394-.415-.554-.159L1.287 16.541a.3.3 0 0 0 .255.459H8.2a.3.3 0 0 0 .3-.3zm3 0V6.046c0-.301.395-.415.555-.159l6.659 10.654a.3.3 0 0 1-.255.459H11.8a.3.3 0 0 1-.3-.3z'/></symbol><symbol viewBox='0 0 40 40' id='uc-icon-original' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M0 40 40 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-rotate' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M13.5.4 12 2l1.5 1.6M12.023 2H14.4A3.6 3.6 0 0 1 18 5.6V8M4 17h9a1 1 0 0 0 1-1V7a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-sad' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M2 17c4.418-4 11.582-4 16 0M16.5 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm-11 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0z'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-saturation' xmlns='http://www.w3.org/2000/svg'><defs><linearGradient id='ouc-icon-id__a' x1='10.001' y1='1' x2='10.001' y2='19' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient></defs><circle cx='10.001' cy='10' r='9' transform='rotate(90 10 10)' fill='url(#ouc-icon-id__a)'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-slider' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M0 10h11m0 0a2 2 0 1 0 4 0m-4 0a2 2 0 1 1 4 0m0 0h5'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-tuning' xmlns='http://www.w3.org/2000/svg'><path stroke-linejoin='round' fill='none' stroke='currentColor' stroke-width='1.2' d='M8 10h11M1 10h4M1 4.5h11m3 0h4m-18 11h11m3 0h4m-7-11a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0M5 10a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0m7 5.5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 1 0-3 0'/></symbol><symbol viewBox='0 0 20 20' id='uc-icon-vibrance' xmlns='http://www.w3.org/2000/svg'><path d='M2.125 5.64A8.96 8.96 0 0 0 1.001 10a8.96 8.96 0 0 0 1.124 4.36V5.64z' fill='url(#ruc-icon-id__a)'/><path d='M2.875 15.499V4.502a9.053 9.053 0 0 1 1.75-1.72v14.437a9.05 9.05 0 0 1-1.75-1.72z' fill='url(#ruc-icon-id__b)'/><path d='M5.375 17.722c.548.33 1.134.601 1.75.809V1.469a8.956 8.956 0 0 0-1.75.81v15.443z' fill='url(#ruc-icon-id__c)'/><path d='M7.875 1.253v17.495c.564.136 1.15.22 1.75.244V1.008a9 9 0 0 0-1.75.245z' fill='url(#ruc-icon-id__d)'/><path d='M10.375 1.008v17.984a9 9 0 0 0 1.75-.244V1.252a9 9 0 0 0-1.75-.244z' fill='url(#ruc-icon-id__e)'/><path d='M12.875 1.469V18.53a8.957 8.957 0 0 0 1.75-.808V2.277a8.957 8.957 0 0 0-1.75-.808z' fill='url(#ruc-icon-id__f)'/><path d='M15.375 2.78v14.44a9.053 9.053 0 0 0 1.75-1.72v-11a9.054 9.054 0 0 0-1.75-1.72z' fill='url(#ruc-icon-id__g)'/><path d='M17.875 5.638v8.724A8.959 8.959 0 0 0 19.001 10a8.96 8.96 0 0 0-1.126-4.362z' fill='url(#ruc-icon-id__h)'/><defs><linearGradient id='ruc-icon-id__a' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__b' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__c' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__d' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__e' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__f' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__g' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient><linearGradient id='ruc-icon-id__h' x1='19.001' y1='10' x2='1.001' y2='10' gradientUnits='userSpaceOnUse'><stop stop-color='#DE15FF'/><stop offset='.203' stop-color='#0029FF'/><stop offset='.479' stop-color='#2AE4F0'/><stop offset='.604' stop-color='#15EF11'/><stop offset='.75' stop-color='#FAE528'/><stop offset='1' stop-color='#EB2A2A'/></linearGradient></defs></symbol><symbol viewBox='0 0 20 20' id='uc-icon-warmth' xmlns='http://www.w3.org/2000/svg'><path d='m7.5 13.05.429.42.171-.175v-.244h-.6zm5 0h-.6v.245l.172.175.428-.42zM8.1 3.5c0-1.05.85-1.9 1.9-1.9V.4a3.1 3.1 0 0 0-3.1 3.1h1.2zm0 9.55V3.5H6.9v9.55h1.2zm-1 2.45c0-.79.315-1.506.829-2.03l-.858-.84A4.088 4.088 0 0 0 5.9 15.5h1.2zm2.9 2.9a2.9 2.9 0 0 1-2.9-2.9H5.9a4.1 4.1 0 0 0 4.1 4.1v-1.2zm2.9-2.9a2.9 2.9 0 0 1-2.9 2.9v1.2a4.1 4.1 0 0 0 4.1-4.1h-1.2zm-.829-2.03c.514.524.829 1.24.829 2.03h1.2c0-1.117-.447-2.13-1.171-2.87l-.858.84zM11.9 3.5v9.55h1.2V3.5h-1.2zM10 1.6c1.05 0 1.9.85 1.9 1.9h1.2A3.1 3.1 0 0 0 10 .4v1.2z' fill='currentColor'/><path d='M10 14V8' stroke='currentColor' stroke-width='1.2' stroke-linecap='round'/><path d='M14 3h3m-3 3h3m-3 3h3m-8 6.5a1 1 0 1 0 2 0 1 1 0 1 0-2 0' stroke='currentColor' stroke-width='1.2'/></symbol></svg>
  <div class="wrapper wrapper_desktop">
    <lr-presence-toggle class="network_problems_splash" set="visible: presence.networkProblems;">
      <div class="network_problems_content">
        <div class="network_problems_icon">
          <lr-icon name="sad"></lr-icon>
        </div>
        <div class="network_problems_text">Network error</div>
      </div>
      <div class="network_problems_footer">
        <lr-btn-ui theme="primary" text="Retry" set="onclick: *on.retryNetwork"></lr-btn-ui>
      </div>
    </lr-presence-toggle>
    <div class="viewport">
      <div class="file_type_outer">
        <div class="file_type">{{fileType}}</div>
      </div>
      <div class="image_container" ref="img-container-el">
        <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=" class="image image_visible_from_editor" ref="img-el" />
        <lr-editor-image-cropper ref="cropper-el"></lr-editor-image-cropper>
        <lr-editor-image-fader ref="fader-el"></lr-editor-image-fader>
      </div>
      <div class="info_pan">{{msg}}</div>
    </div>
    <div class="toolbar">
      <lr-line-loader-ui set="active: showLoader"></lr-line-loader-ui>
      <div class="toolbar_content toolbar_content__editor">
        <lr-editor-toolbar></lr-editor-toolbar>
      </div>
    </div>
  </div>
`;
  static observedAttributes = ["uuid", "cdn-url", "crop-preset", "tabs"];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Config = class {
  static observedAttributes = [
    "debug",
    "pubkey",
    "multiple",
    "multiple-min",
    "multiple-max",
    "confirm-upload",
    "img-only",
    "accept",
    "external-sources-preferred-types",
    "store",
    "camera-mirror",
    "camera-capture",
    "source-list",
    "cloud-image-editor-tabs",
    "max-local-file-size-bytes",
    "thumb-size",
    "show-empty-list",
    "use-local-image-editor",
    "use-cloud-image-editor",
    "remove-copyright",
    "crop-preset",
    "image-shrink",
    "modal-scroll-lock",
    "modal-backdrop-strokes",
    "source-list-wrap",
    "remote-tab-session-key",
    "cdn-cname",
    "base-url",
    "social-base-url",
    "secure-signature",
    "secure-expire",
    "secure-delivery-proxy",
    "retry-throttled-request-max-times",
    "multipart-min-file-size",
    "multipart-chunk-size",
    "max-concurrent-requests",
    "multipart-max-concurrent-requests",
    "multipart-max-attempts",
    "check-for-url-duplicates",
    "save-url-for-recurrent-uploads",
    "group-output",
    "user-agent-integration",
    "locale-name",
    "multiplemin",
    "multiplemax",
    "confirmupload",
    "imgonly",
    "externalsourcespreferredtypes",
    "cameramirror",
    "cameracapture",
    "sourcelist",
    "cloudimageeditortabs",
    "maxlocalfilesizebytes",
    "thumbsize",
    "showemptylist",
    "uselocalimageeditor",
    "usecloudimageeditor",
    "removecopyright",
    "croppreset",
    "imageshrink",
    "modalscrolllock",
    "modalbackdropstrokes",
    "sourcelistwrap",
    "remotetabsessionkey",
    "cdncname",
    "baseurl",
    "socialbaseurl",
    "securesignature",
    "secureexpire",
    "securedeliveryproxy",
    "retrythrottledrequestmaxtimes",
    "multipartminfilesize",
    "multipartchunksize",
    "maxconcurrentrequests",
    "multipartmaxconcurrentrequests",
    "multipartmaxattempts",
    "checkforurlduplicates",
    "saveurlforrecurrentuploads",
    "groupoutput",
    "useragentintegration",
    "localename",
  ];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const ConfirmationDialog = class {
  static template = `
  <lr-activity-header>
    <button type="button" class="mini-btn" set="onclick: *historyBack">
      <lr-icon name="back"></lr-icon>
    </button>
    <span>{{activityCaption}}</span>
    <button type="button" class="mini-btn close-btn" set="onclick: *closeModal">
      <lr-icon name="close"></lr-icon>
    </button>
  </lr-activity-header>

  <div class="message">{{messageTxt}}</div>
  <div class="toolbar">
    <button type="button" class="deny-btn secondary-btn" set="onclick: onDeny">{{denyBtnTxt}}</button>
    <button type="button" class="confirm-btn primary-btn" set="onclick: onConfirm">{{confirmBtnTxt}}</button>
  </div>
`;
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Copyright = class {
  static template = `
    <a
      href="https://uploadcare.com/?utm_source=copyright&utm_medium=referral&utm_campaign=v4"
      target="_blank noopener"
      class="credits"
      >Powered by Uploadcare</a
    >
  `;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const CropFrame = class {
  static template = ` <svg class="svg" ref="svg-el" xmlns="http://www.w3.org/2000/svg"></svg> `;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Data = class {
  static warn = () => {};
  static registerCtx = () => {};
  static deleteCtx = () => {};
  static getCtx = () => {};
  static globalStore = {};
  static apply = () => {};
  static bind = () => {};
  static call = () => {};
  static toString = () => {};
  static hasOwnProperty = () => {};
  static isPrototypeOf = () => {};
  static propertyIsEnumerable = () => {};
  static valueOf = () => {};
  static toLocaleString = () => {};
};
export const DropArea = class {
  static template = `
  <slot>
    <div data-default-slot hidden></div>
    <div ref="content-wrapper" class="content-wrapper" set="@hidden: !isVisible">
      <div class="icon-container" set="@hidden: !withIcon">
        <lr-icon name="default"></lr-icon>
        <lr-icon name="arrow-down"></lr-icon>
      </div>
      <span class="text">{{text}}</span>
    </div>
  </slot>
`;
  static observedAttributes = [
    "with-icon",
    "clickable",
    "text",
    "fullscreen",
    "disabled",
  ];
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorCropButtonControl = class {
  static template = `
  <div class="before"></div>
  <lr-icon set="@name: icon;"></lr-icon>
  <div class="title" ref="title-el">{{title}}</div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorFilterControl = class {
  static template = `
  <div class="before"></div>
  <lr-icon set="@name: icon;"></lr-icon>
  <div class="title" ref="title-el">{{title}}</div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorImageCropper = class {
  static template = `
  <canvas class="canvas" ref="canvas-el"></canvas>
  <lr-crop-frame ref="frame-el"></lr-crop-frame>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorImageFader = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorOperationControl = class {
  static template = `
  <div class="before"></div>
  <lr-icon set="@name: icon;"></lr-icon>
  <div class="title" ref="title-el">{{title}}</div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorScroller = class {
  static template = ` <slot></slot> `;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorSlider = class {
  static template = `
  <lr-slider-ui
    ref="slider-el"
    set="disabled: disabled; min: min; max: max; defaultValue: defaultValue; zero: zero; onInput: on.input;"
  ></lr-slider-ui>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const EditorToolbar = class {
  static template = `
  <lr-line-loader-ui set="active: showLoader"></lr-line-loader-ui>
  <div class="info-tooltip_container">
    <div class="info-tooltip_wrapper">
      <div ref="tooltip-el" class="info-tooltip info-tooltip_hidden">{{*operationTooltip}}</div>
    </div>
  </div>
  <div class="toolbar-container">
    <lr-presence-toggle class="sub-toolbar" set="visible: presence.mainToolbar; styles: presence.subTopToolbarStyles">
      <div class="tab-content-row">
    <lr-presence-toggle class="tab-content" set="visible: presence.tabContent.crop; styles: presence.tabContentStyles">
      <lr-editor-scroller hidden-scrollbar>
        <div class="controls-list_align">
          <div class="controls-list_inner" ref="controls-list-crop"></div>
        </div>
      </lr-editor-scroller>
    </lr-presence-toggle>
  
    <lr-presence-toggle class="tab-content" set="visible: presence.tabContent.tuning; styles: presence.tabContentStyles">
      <lr-editor-scroller hidden-scrollbar>
        <div class="controls-list_align">
          <div class="controls-list_inner" ref="controls-list-tuning"></div>
        </div>
      </lr-editor-scroller>
    </lr-presence-toggle>
  
    <lr-presence-toggle class="tab-content" set="visible: presence.tabContent.filters; styles: presence.tabContentStyles">
      <lr-editor-scroller hidden-scrollbar>
        <div class="controls-list_align">
          <div class="controls-list_inner" ref="controls-list-filters"></div>
        </div>
      </lr-editor-scroller>
    </lr-presence-toggle>
  </div>
      <div class="controls-row">
        <lr-btn-ui theme="boring" icon="closeMax" set="onclick: on.cancel"> </lr-btn-ui>
        <lr-presence-toggle class="tab-toggles" set="visible: presence.tabToggles; styles: presence.tabTogglesStyles">
          <div ref="tabs-indicator" class="tab-toggles_indicator"></div>
          
    <lr-presence-toggle class="tab-toggle" set="visible: presence.tabToggle.crop; styles: presence.tabToggleStyles;">
      <lr-btn-ui
        theme="boring"
        ref="tab-toggle-crop"
        data-id="crop"
        icon="crop"
        tabindex="0"
        set="onclick: on.clickTab;"
      >
      </lr-btn-ui>
    </lr-presence-toggle>
  
    <lr-presence-toggle class="tab-toggle" set="visible: presence.tabToggle.tuning; styles: presence.tabToggleStyles;">
      <lr-btn-ui
        theme="boring"
        ref="tab-toggle-tuning"
        data-id="tuning"
        icon="tuning"
        tabindex="0"
        set="onclick: on.clickTab;"
      >
      </lr-btn-ui>
    </lr-presence-toggle>
  
    <lr-presence-toggle class="tab-toggle" set="visible: presence.tabToggle.filters; styles: presence.tabToggleStyles;">
      <lr-btn-ui
        theme="boring"
        ref="tab-toggle-filters"
        data-id="filters"
        icon="filters"
        tabindex="0"
        set="onclick: on.clickTab;"
      >
      </lr-btn-ui>
    </lr-presence-toggle>
  
        </lr-presence-toggle>
        <lr-btn-ui theme="primary" icon="done" set="onclick: on.apply"> </lr-btn-ui>
      </div>
    </lr-presence-toggle>
    <lr-presence-toggle class="sub-toolbar" set="visible: presence.subToolbar; styles: presence.subBottomToolbarStyles">
      <div class="slider">
        <lr-editor-slider ref="slider-el"></lr-editor-slider>
      </div>
      <div class="controls-row">
        <lr-btn-ui theme="boring" set="onclick: on.cancelSlider;" l10n="@text:cancel"> </lr-btn-ui>
        <lr-btn-ui theme="primary" set="onclick: on.applySlider;" l10n="@text:apply"> </lr-btn-ui>
      </div>
    </lr-presence-toggle>
  </div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const ExternalSource = class {
  static template = `
  <lr-activity-header>
    <button type="button" class="mini-btn" set="onclick: *historyBack">
      <lr-icon name="back"></lr-icon>
    </button>
    <div>
      <lr-icon set="@name: activityIcon"></lr-icon>
      <span>{{activityCaption}}</span>
    </div>
    <button type="button" class="mini-btn close-btn" set="onclick: *historyBack">
      <lr-icon name="close"></lr-icon>
    </button>
  </lr-activity-header>
  <div class="content">
    <div ref="iframeWrapper" class="iframe-wrapper"></div>
    <div class="toolbar">
      <button type="button" class="cancel-btn secondary-btn" set="onclick: onCancel" l10n="cancel"></button>
      <div></div>
      <div set="@hidden: !multiple" class="selected-counter"><span l10n="selected-count"></span>{{counter}}</div>
      <button type="button" class="done-btn primary-btn" set="onclick: onDone; @disabled: !counter">
        <lr-icon name="check"></lr-icon>
      </button>
    </div>
  </div>
`;
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FileItem = class {
  static template = `
  <div class="inner" set="@finished: isFinished; @uploading: isUploading; @failed: isFailed; @focused: isFocused">
    <div class="thumb" set="style.backgroundImage: thumbUrl">
      <div class="badge">
        <lr-icon set="@name: badgeIcon"></lr-icon>
      </div>
    </div>
    <div class="file-name-wrapper">
      <span class="file-name" set="@title: itemName">{{itemName}}</span>
      <span class="file-error" set="@hidden: !errorText">{{errorText}}</span>
    </div>
    <div class="file-actions">
      <button type="button" class="edit-btn mini-btn" set="onclick: onEdit; @hidden: !isEditable">
        <lr-icon name="edit-file"></lr-icon>
      </button>
      <button type="button" class="remove-btn mini-btn" set="onclick: onRemove;">
        <lr-icon name="remove-file"></lr-icon>
      </button>
      <button type="button" class="upload-btn mini-btn" set="onclick: onUpload;">
        <lr-icon name="upload"></lr-icon>
      </button>
    </div>
    <lr-progress-bar class="progress-bar" set="value: progressValue; visible: progressVisible;"> </lr-progress-bar>
  </div>
`;
  static activeInstances = {};
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FilePreview = class {
  static template = ` <lr-img class="img-view" ref="img" set="@src: src; style.aa: src;" /> `;
  static observedAttributes = ["checkerboard"];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FileUploaderInline = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FileUploaderMinimal = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FileUploaderRegular = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const FormInput = class {
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Icon = class {
  static template = `
  <svg ref="svg" xmlns="http://www.w3.org/2000/svg">
    <title>{{title}}</title>
    <use set="@href: href;"></use>
  </svg>
`;
  static observedAttributes = ["name", "title"];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Img = class {
  static observedAttributes = [
    "dev-mode",
    "pubkey",
    "uuid",
    "src",
    "lazy",
    "intersection",
    "breakpoints",
    "cdn-cname",
    "proxy-cname",
    "secure-delivery-proxy",
    "hi-res-support",
    "ultra-res-support",
    "format",
    "cdn-operations",
    "progressive",
    "quality",
    "is-background-for",
    "is-preview-blur",
  ];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const LineLoaderUi = class {
  static template = `
  <div class="inner">
    <div class="line" ref="line-el"></div>
  </div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const LrBtnUi = class {
  static observedAttributes = ["text", "icon", "reverse", "theme"];
  static template = `
  <lr-icon set="className: iconCss; @name: icon; @hidden: !icon"></lr-icon>
  <div class="text">{{text}}</div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Modal = class {
  static StateConsumerScope = `modal`;
  static template = `
  <dialog ref="dialog">
    <slot></slot>
  </dialog>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const PACKAGE_NAME = `blocks`;
export const PACKAGE_VERSION = `0.40.0`;
export const PresenceToggle = class {
  static template = ` <slot></slot> `;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const ProgressBar = class {
  static template = ` <div ref="line" class="progress"></div> `;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const ProgressBarCommon = class {
  static template = ` <lr-progress-bar set="visible: visible; value: value"></lr-progress-bar> `;
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Select = class {
  static template = `
  <button>
    {{currentText}}
    <lr-icon name="select"></lr-icon>
    <select ref="select" set="innerHTML: selectHtml; onchange: onSelect"></select>
  </button>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const SimpleBtn = class {
  static template = `
  <lr-drop-area set="@disabled: !withDropZone">
    <button type="button" set="onclick: onClick">
      <lr-icon name="upload"></lr-icon>
      <span l10n="button-text"></span>
      <slot></slot>
      <div class="visual-drop-area" l10n="drop-files-here"></div>
    </button>
  </lr-drop-area>
`;
  static observedAttributes = ["dropzone"];
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const SliderUi = class {
  static template = `
  <div class="steps" ref="steps-el"></div>
  <div ref="thumb-el" class="thumb"></div>
  <input
    class="input"
    type="range"
    ref="input-el"
    tabindex="0"
    set="oninput: on.sliderInput; onchange: on.sliderChange; @min: min; @max: max; @value: defaultValue;"
  />
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const SourceBtn = class {
  static template = `
  <lr-icon set="@name: iconName"></lr-icon>
  <div class="txt" l10n="src-type"></div>
`;
  static observedAttributes = ["type"];
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const SourceList = class {
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const StartFrom = class {
  static template = ` <div class="content"><slot></slot></div> `;
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Tabs = class {
  static observedAttributes = ["tab-list", "default"];
  static template = `
  <div ref="row" class="tabs-row"></div>
  <div ref="context" class="tabs-context">
    <slot></slot>
  </div>
`;
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const UID = class {
  static generate = () => {};
  static apply = () => {};
  static bind = () => {};
  static call = () => {};
  static toString = () => {};
  static hasOwnProperty = () => {};
  static isPrototypeOf = () => {};
  static propertyIsEnumerable = () => {};
  static valueOf = () => {};
  static toLocaleString = () => {};
};
export const UploadCtxProvider = class {
  static EventType = {
    FILE_ADDED: "file-added",
    FILE_REMOVED: "file-removed",
    FILE_UPLOAD_START: "file-upload-start",
    FILE_UPLOAD_PROGRESS: "file-upload-progress",
    FILE_UPLOAD_SUCCESS: "file-upload-success",
    FILE_UPLOAD_FAILED: "file-upload-failed",
    FILE_URL_CHANGED: "file-url-changed",
    MODAL_OPEN: "modal-open",
    MODAL_CLOSE: "modal-close",
    DONE_CLICK: "done-click",
    UPLOAD_CLICK: "upload-click",
    ACTIVITY_CHANGE: "activity-change",
    COMMON_UPLOAD_START: "common-upload-start",
    COMMON_UPLOAD_PROGRESS: "common-upload-progress",
    COMMON_UPLOAD_SUCCESS: "common-upload-success",
    COMMON_UPLOAD_FAILED: "common-upload-failed",
    CHANGE: "change",
    GROUP_CREATED: "group-created",
  };
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const UploadList = class {
  static template = `
  <lr-activity-header>
    <span class="header-text">{{headerText}}</span>
    <button type="button" class="mini-btn close-btn" set="onclick: *closeModal">
      <lr-icon name="close"></lr-icon>
    </button>
  </lr-activity-header>

  <div class="no-files" set="@hidden: hasFiles">
    <slot name="empty"><span l10n="no-files"></span></slot>
  </div>

  <div class="files" repeat="*uploadList" repeat-item-tag="lr-file-item"></div>

  <div class="common-error" set="@hidden: !commonErrorMessage; textContent: commonErrorMessage;"></div>

  <div class="toolbar">
    <button type="button" class="cancel-btn secondary-btn" set="onclick: onCancel;" l10n="clear"></button>
    <div class="toolbar-spacer"></div>
    <button
      type="button"
      class="add-more-btn secondary-btn"
      set="onclick: onAdd; @disabled: !addMoreBtnEnabled; @hidden: !addMoreBtnVisible"
    >
      <lr-icon name="add"></lr-icon><span l10n="add-more"></span>
    </button>
    <button
      type="button"
      class="upload-btn primary-btn"
      set="@hidden: !uploadBtnVisible; onclick: onUpload;"
      l10n="upload"
    ></button>
    <button
      type="button"
      class="done-btn primary-btn"
      set="@hidden: !doneBtnVisible; onclick: onDone;  @disabled: !doneBtnEnabled"
      l10n="done"
    ></button>
  </div>

  <lr-drop-area ghost></lr-drop-area>
`;
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const UploaderBlock = class {
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const UrlSource = class {
  static template = `
  <lr-activity-header>
    <button type="button" class="mini-btn" set="onclick: *historyBack">
      <lr-icon name="back"></lr-icon>
    </button>
    <div>
      <lr-icon name="url"></lr-icon>
      <span l10n="caption-from-url"></span>
    </div>
    <button type="button" class="mini-btn close-btn" set="onclick: *closeModal">
      <lr-icon name="close"></lr-icon>
    </button>
  </lr-activity-header>
  <form class="content">
    <input placeholder="https://" class="url-input" type="text" ref="input" set="oninput: onInput" />
    <button
      type="submit"
      class="url-upload-btn primary-btn"
      set="onclick: onUpload; @disabled: importDisabled"
      l10n="upload-url"
    ></button>
  </form>
`;
  static extSrcList = {
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static sourceTypes = {
    LOCAL: "local",
    URL: "url",
    CAMERA: "camera",
    DRAW: "draw",
    FACEBOOK: "facebook",
    DROPBOX: "dropbox",
    GDRIVE: "gdrive",
    GPHOTOS: "gphotos",
    INSTAGRAM: "instagram",
    FLICKR: "flickr",
    VK: "vk",
    EVERNOTE: "evernote",
    BOX: "box",
    ONEDRIVE: "onedrive",
    HUDDLE: "huddle",
  };
  static activities = {
    START_FROM: "start-from",
    CAMERA: "camera",
    DRAW: "draw",
    UPLOAD_LIST: "upload-list",
    URL: "url",
    CONFIRMATION: "confirmation",
    CLOUD_IMG_EDIT: "cloud-image-edit",
    EXTERNAL: "external",
    DETAILS: "details",
  };
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const Video = class {
  static template = `
  <div class="video-wrapper">
    <video ref="video" preload="metadata" crossorigin="anonymous"></video>
  </div>

  <div class="toolbar">
    <div class="progress" ref="progress" set -onclick="progressClicked">
      <div class="bar" set -style.width="progressCssWidth"></div>
    </div>

    <div class="tb-block">
      <button set -onclick="onPP">
        <lr-icon set -@name="ppIcon"></lr-icon>
      </button>
      <div class="timer">{{currentTime}} / {{totalTime}}</div>
    </div>

    <div class="tb-block">
      <button set -onclick="onCap" -@hidden="!hasSubtitles">
        <lr-icon set -@name="capIcon"></lr-icon>
      </button>

      <button set -onclick="onMute">
        <lr-icon set -@name="volIcon"></lr-icon>
      </button>

      <lr-range type="range" set -onchange="onVolChange" -@disabled="volumeDisabled" -value="volumeValue"> </lr-range>

      <button set -onclick="onFs">
        <lr-icon set -@name="fsIcon"></lr-icon>
      </button>
    </div>
  </div>
`;
  static observedAttributes = ["video", "src"];
  static reg = () => {};
  static is = `sym-1`;
  static bindAttributes = () => {};
};
export const connectBlocksFrom = () => {};
export const defineLocale = () => {};
export const registerBlocks = () => {};
export const toKebabCase = () => {};
